### PR TITLE
General refactor of the database classes

### PR DESCRIPTION
**Pull request by leonardvaney**
In the time left I have to spend, this is the objectives of this PR:

- [x] Delete the classes ExistingPicture, NewPicture, ReportedPicture and Picture (and use instead the database classes)
- [x] Remove unsused functions from the User classes and simplify what is left (if possible)
- [x] Remove as many code climate issues as I can from the database classes

### CODE REVIEW

File: app/src/androidTest/java/com/github/wnder/FirebaseUserDatabaseTest.java [hunk 1]
```diff
@@ -28,11 +28,12 @@
     public void gettingNewPicturesReturnAPictureNotGuessed(){
         GlobalUser.setUser(new SignedInUser("testUser", Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag)));
         User user = GlobalUser.getUser();
+        user.setRadius(100000);
```
Comment (amattei777) [inline app/src/androidTest/java/com/github/wnder/FirebaseUserDatabaseTest.java:31]: magic number

**Review by amattei777 — CHANGES_REQUESTED** [review]
good enough for this time of the day

File: app/src/androidTest/java/com/github/wnder/FirebaseUserDatabaseTest.java (hunk 1 above)
Comment (leonardvaney) [reply app/src/androidTest/java/com/github/wnder/FirebaseUserDatabaseTest.java]: It's done.

File: app/src/main/java/com/github/wnder/picture/Picture.java [hunk 2]
```diff
@@ -1,158 +1,14 @@
...
-import java.util.TreeMap;
-import java.util.function.Consumer;
-
 /**
  * abstract class for all kind of pictures
  */
-public abstract class Picture {
+public final class Picture {
 
     public final static String UNINITIALIZED_ID = "";
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/picture/Picture.java:8]: You should find another place to put this string, maybe we could just use a class with all the "public static final" strings that we use often/in a lot of places though I don't know if it's good practice.

File: app/src/androidTest/java/com/github/wnder/FirebaseUserDatabaseTest.java [hunk 3]
```diff
@@ -23,19 +23,19 @@
...
         List<String> guessedPics = new ArrayList<>();
         String receivedPic = "";
         try {
             guessedPics = db.getPictureList(user, "guessedPics").get();
-            receivedPic = db.getNewPictureForUser(user, 100000).get();
-        } catch (ExecutionException e) {
-            e.printStackTrace();
-        } catch (InterruptedException e) {
+            receivedPic = db.getNewPictureForUser(user).get();
+        } catch (ExecutionException | InterruptedException e) {
```
Comment (rkuenzi-epfl) [inline app/src/androidTest/java/com/github/wnder/FirebaseUserDatabaseTest.java:38]: If it comes to this you could just use `Exception e` (for next time).

File: app/src/main/java/com/github/wnder/GuessPreviewActivity.java [hunk 4]
```diff
@@ -125,26 +126,21 @@ protected void onStart() {
...
+                //If not, display default picture
+                // Maybe create a bitmap that tells that no pictures were available (this one is just the one available)
+                Bitmap bmp = BitmapFactory.decodeResource(getResources(), R.raw.no_image);
+                setImageViewBitmap(bmp, picId);
             }
-            );
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
+        });
```
Comment (rkuenzi-epfl) [inline app/src/main/java/com/github/wnder/GuessPreviewActivity.java:143]: You could use the occassion to properly handle the exception for `getNewPictureForUser` (with .exceptionnaly) for example showing the "no image" picture.

**Review by rkuenzi-epfl — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/wnder/picture/Picture.java (hunk 2 above)
Comment (leonardvaney) [reply app/src/main/java/com/github/wnder/picture/Picture.java]: I don't know either but for now it should be at least placed in a class with a better name, so I will do that.

**Review by rkuenzi-epfl — APPROVED** [review]
All good

**Review by amattei777 — APPROVED** [review]

**Pull request merged**